hardware/battery_management_system_protocol.ksy

```diff
@@ -0,0 +1,269 @@
+meta:
+  id: battery_management_system_protocol
+  title: Communication protocol of smart battery management systems from LLT power
+  license: CC0-1.0
+  ks-version: 0.9
+  endian: be
+  bit-endian: be
+
+doc: |
+  Many modern general purpose BMS include a UART/Bluetooth based communication interface.
+  After sending read requests they respond with various information's about the battery state in
+  a custom binary format.
+
+doc-ref: https://www.lithiumbatterypcb.com/Protocol%20English%20Version.rar
+
+seq:
+  - id: magic_start
+    contents: [0xdd]
+  - id: cmd
+    type: u1
+    -affected-by: 778
+
+  - size: 0
+    if: ofs_body_start < 0 # storing current position
+  - id: body
+    type:
+      switch-on: cmd
+      cases:
+        commands::read.to_i: read_req
+        commands::write.to_i: write_req
+        _: response(cmd)
+  - size: 0
+    if: ofs_body_end < 0 # storing current position
+
+  - id: checksum
+    type: u2
+    doc: |
+      Should be equal to the result from: 0x10000 - sum(body)
+      Where sum() is calculated over the value of every byte individually. This works
+      for up to 256 bytes, for more it must be clamped to 16 bits: result & 0xffff
+      Body includes everything besides magic start/end byte, cmd and checksum,
+      so excluding 2 bytes at the beginning and 3 bytes at the end.
```

Comment:
You don't need to write it here, it is sufficiently specified by the `checksum_input` attribute. Anyone can simply access `checksum_input` and they will directly obtain the byte array that's needed to calculate the checksum. It is also apparent from the spec where these bytes come from.